### ISSUE TITLE
Python syntax to specify options for planCovPath and planCovRoute

### ISSUE THREAD

**Issue by aosmw** [Fields2Cover/Fields2Cover#145]
TLDR; Syntax to access Options for planCovPath and planCovRoute from python?

Hi here are some fun routes/paths generated using python with `planCovPath(robot, field, local_crs)` and `planCovRoute(robot, field, local_crs)` using python based on tutorials/python/quick_start.py  

Playing around with robot coverage width 2.0m and 6.0m, turning circle 3.0m.

For the smaller coverage width, 2.0, which results in loopy turns, I'd like to play around with the Route planning algorithms (snake, spiral)

I don't know how to follow the python swig magic. ... Yet.

What is the syntax to be able to access/specify `Options` to use the 
F2CPath planCovPath(const F2CRobot& robot, const F2CField& field, const Options& opt = Options(), bool local_crs = true);  from python?
https://github.com/Fields2Cover/Fields2Cover/blob/fc35541c0d3b4222ef1ec86507a608e2682ebe7b/include/fields2cover.h#L135

###  robot coverage path -  coverage width 2.0m turning circle 3.0m
![cov_path 2m](https://github.com/Fields2Cover/Fields2Cover/assets/116058035/86c74ff0-4045-46d3-a2b3-7ac5b028d2a7)

### robot coverage path - coverage width 6.0m turning circle 3.0m
![cov_path 6m](https://github.com/Fields2Cover/Fields2Cover/assets/116058035/f19c9ec7-ae7a-4655-b6eb-a00c6b7d8669)

**Comment by aosmw** [comment]
swig produces build/fields2cover/swig/python/fields2cover.py which creates this snippet.

```
SGAlg_BRUTE_FORCE = _fields2cover_python.SGAlg_BRUTE_FORCE                                                                                                                                    SGAlg_GIVEN_ANGLE = _fields2cover_python.SGAlg_GIVEN_ANGLE                                                                                                                                    SGObjFunc_N_SWATH_MOD = _fields2cover_python.SGObjFunc_N_SWATH_MOD                                                                                                                            SGObjFunc_N_SWATH = _fields2cover_python.SGObjFunc_N_SWATH                                                                                                                                    SGObjFunc_SWATH_LENGTH = _fields2cover_python.SGObjFunc_SWATH_LENGTH                                                                                                                          SGObjFunc_FIELD_COV = _fields2cover_python.SGObjFunc_FIELD_COV                                                                                                                                SGObjFunc_OVERLAPS = _fields2cover_python.SGObjFunc_OVERLAPS                                                                                                                                  RPAlg_SHORTEST_ROUTE = _fields2cover_python.RPAlg_SHORTEST_ROUTE                                                                                                                              PPAlg_DUBINS = _fields2cover_python.PPAlg_DUBINS                                                                                                                                              PPAlg_DUBINS_CC = _fields2cover_python.PPAlg_DUBINS_CC                                                                                                                                        PPAlg_REEDS_SHEPP = _fields2cover_python.PPAlg_REEDS_SHEPP                                                                                                                                    PPAlg_REEDS_SHEPP_HC = _fields2cover_python.PPAlg_REEDS_SHEPP_HC                                                                                                                              HGAlg_NONE = _fields2cover_python.HGAlg_NONE                                                                                                                                                  HGAlg_CONST = _fields2cover_python.HGAlg_CONST                                                                                                                                                DecompAlg_NONE = _fields2cover_python.DecompAlg_NONE                                                                                                                                          class Options(object):                                                                                                                                                                            thisown = property(lambda x: x.this.own(), lambda x, v: x.this.own(v), doc="The membership flag")                                                                                             __repr__ = _swig_repr                                                                                                                                                                         decomp_alg = property(_fields2cover_python.Options_decomp_alg_get, _fields2cover_python.Options_decomp_alg_set)                                                                               decomp_angle = property(_fields2cover_python.Options_decomp_angle_get, _fields2cover_python.Options_decomp_angle_set)                                                                         hg_alg = property(_fields2cover_python.Options_hg_alg_get, _fields2cover_python.Options_hg_alg_set)                                                                                           hg_swaths = property(_fields2cover_python.Options_hg_swaths_get, _fields2cover_python.Options_hg_swaths_set)                                                                                  sg_alg = property(_fields2cover_python.Options_sg_alg_get, _fields2cover_python.Options_sg_alg_set)                                                                                           sg_obj = property(_fields2cover_python.Options_sg_obj_get, _fields2cover_python.Options_sg_obj_set)                                                                                           sg_angle = property(_fields2cover_python.Options_sg_angle_get, _fields2cover_python.Options_sg_angle_set)                                                                                     rp_alg = property(_fields2cover_python.Options_rp_alg_get, _fields2cover_python.Options_rp_alg_set)                                                                                           pp_alg = property(_fields2cover_python.Options_pp_alg_get, _fields2cover_python.Options_pp_alg_set)                                                                                                                                                                                                                                                                                         def __init__(self):                                                                                                                                                                               _fields2cover_python.Options_swiginit(self, _fields2cover_python.new_Options())                                                                                                           __swig_destroy__ = _fields2cover_python.delete_Options                                                                                                                                                                                                                                                                                                                                  # Register Options in _fields2cover_python:                                                                                                                                                   _fields2cover_python.Options_swigregister(Options)
```

**Comment by Gonzalo-Mier** [comment]
Hi @aosmw,
This class is not well tested, nor well implemented yet and that's why it's not on the tutorials. On the other hand, feel free to explore it but please understand it is not properly working.

The code to that class is here:
https://github.com/Fields2Cover/Fields2Cover/blob/fc35541c0d3b4222ef1ec86507a608e2682ebe7b/include/fields2cover.h#L70-L168
https://github.com/Fields2Cover/Fields2Cover/blob/fc35541c0d3b4222ef1ec86507a608e2682ebe7b/src/fields2cover.cpp#L10-L164

And swig magic is declared here: 
https://github.com/Fields2Cover/Fields2Cover/blob/fc35541c0d3b4222ef1ec86507a608e2682ebe7b/swig/Fields2Cover.i

To access to those options, first declare the class and then give your values to it:
```
opt = f2c.Options()
opt.sg_alg = f2c.SGAlg_GIVEN_ANGLE
```

**Comment by aosmw** [comment]
Thanks @Gonzalo-Mier,

Here is what I am using, to round out the issue.  I think it is selecting the defaults.  I'll play around, noting your caveat.

```
    options = f2c.Options()
    options.decomp_alg = f2c.DecompAlg_NONE
    options.decomp_angle = 0.0
    options.hg_alg = f2c.HGAlg_CONST
    options.hg_swaths = 3
    options.sg_alg = f2c.SGAlg_GIVEN_ANGLE
    options.sg_obj = f2c.SGObjFunc_N_SWATH_MOD
    options.sg_angle = 0.0
    options.rp_alg = f2c.RPAlg_SHORTEST_ROUTE
    options.pp_alg = f2c.PPAlg_DUBINS

    local_crs=False
    cov_route = f2c.planCovRoute(robot, field, options, local_crs)
    #Coverage plan the path
    cov_path = f2c.planCovPath(robot, field, options, local_crs);
```
